includes/functions/llms-functions-content.php

```diff
@@ -15,83 +15,129 @@
+		 * @param array   $restrictions Result from `llms_page_restricted()` for the current post.
 		 */
-		return apply_filters( 'llms_get_post_content', do_shortcode( $before . $content . $after ), $post, $page_restricted );
+		return apply_filters( 'llms_get_post_content', do_shortcode( $before . $content . $after ), $post, $restrictions );
+
+	}
+}
+
+/**
+ * Retrieve the sales page content for a course or membership
+ *
+ * By default only courses and memberships support sales pages, the meta property
+ * must be set to `content` or an empty string, and the post must have a `post_excerpt`
+ * property value.
+ *
+ * @since [version]
+ *
+ * @param WP_Post $post    The post object.
+ * @param string  $default Optional. Default content to use when no override content can be found.
+ * @return string
+ */
+function llms_get_post_sales_page_content( $post, $default = '' ) {
+
+	$content = $default;
+
+	if ( post_type_supports( $post->post_type, 'llms-sales-page' ) ) {
+		$sales_page = get_post_meta( $post->ID, '_llms_sales_page_content_type', true );
+		if ( $post->post_excerpt && ( '' === $sales_page || 'content' === $sales_page ) ) {
+			add_filter( 'the_excerpt', array( $GLOBALS['wp_embed'], 'autoembed' ), 9 );
+			$content = llms_get_excerpt( $post->ID );
+		}
+	}
+
+	/**
+	 * Filters the HTML content of a LifterLMS post type's sales page content
+	 *
+	 * @since [version]
+	 *
+	 * @param string  $content HTML content of the sales page.
+	 * @param WP_Post $content Post object.
+	 * @param string  $default Default content used when no override content can be found.
+	 */
+	return apply_filters( 'llms_post_sales_page_content', $content, $post, $default );
 
+}
+
+/**
+ * Initialize LifterLMS post type content filters
+ *
+ * This method is used to determine whether or `llms_get_post_content()` should automatically
+ * be added as a filter callback for the WP core `the_content` filter.
+ *
+ * When working with posts on the admin panel (during course building, importing) we don't want
+ * other plugins that may desire running `apply_filters( 'the_content', $content )` to apply our
+ * plugin's filters.
+ *
+ * Additionally, we don't want to return template information when processing REST requests.
+ *
+ * @since [version]
+ *
+ * @return boolean Returns `true` if content filters are added and `false` if not.
+ */
+function llms_post_content_init() {
+
+	// Don't filter any requests on the admin panel or when processing REST requests.
+	$should_filter = ( ! is_admin() && ! llms_is_rest() );
+
+	/**
+	 * Filters whether or not LifterLMS content filters should be applied.
+	 *
+	 * @since [version]
+	 *
+	 * @param boolean $should_filter Whether or not to filter the content.
+	 */
+	if ( apply_filters( 'llms_should_filter_post_content', $should_filter ) ) {
+		return add_filter( 'the_content', 'llms_get_post_content' );
 	}
+
+	return false;
+
 }
-add_filter( 'the_content', 'llms_get_post_content' );
+add_action( 'init', 'llms_post_content_init' );
```

Comment:
normally this file is already included at `init|10`:
https://github.com/gocodebox/lifterlms/blob/4.16.0/class-lifterlms.php#L119

Except when that function is called directly, e.g.:
https://github.com/gocodebox/lifterlms/blob/4.16.0/includes/class.llms.emails.php#L62
https://github.com/gocodebox/lifterlms/blob/4.16.0/includes/admin/class.llms.admin.import.php#L284

So maybe this line could be:
`current_filter( 'init' ) ? llms_post_content_init() : add_action( 'init', 'llms_post_content_init' );`

so that we don't alter the original "execution flow".

Reply:
@eri-trabiccolo in both of these manually called scenarios we don't actually care if the filter is loaded though, right?

1) During email loading we don't need to add any of our custom content to the qualifying post types
2) Uploading an import happens on the admin panel, where we wouldn't apply the filter anyway

In both scenarios it's probably preferable not to have the filter loaded from at the very least a minor performance perspective.

Do you agree?

Reply:
@thomasplevy yeah I agree, I was exposing the current scenario.
So we can just go with
`llms_post_content_init()` instead of `add_action( 'init', 'llms_post_content_init' );`